sdk/python/kfp/components/_python_op.py

```diff
@@ -191,6 +191,66 @@ def _capture_function_code_using_cloudpickle(func, modules_to_capture: List[str]
     return function_loading_code
 
 
+def strip_type_hints(source_code: str) -> str:
+    try:
+        return _strip_type_hints_using_strip_hints(source_code)
+    except Exception as ex:
+        print('Error when stripping type annotations: ' + str(ex))
+
+    try:        
+        return _strip_type_hints_using_lib2to3(source_code)
+    except Exception as ex:
+        print('Error when stripping type annotations: ' + str(ex))
+
+    return source_code    
+
+
+def _strip_type_hints_using_strip_hints(source_code: str) -> str:
+    from strip_hints import strip_string_to_string
+
+    # Workaround for https://github.com/abarker/strip-hints/issues/4 , https://bugs.python.org/issue35107
+    # I could not repro it though
+    if source_code[-1] != '\n':
+        source_code += '\n'
+
+    return strip_string_to_string(source_code, to_empty=True)
+
+
+def _strip_type_hints_using_lib2to3(source_code: str) -> str:
+    """Strips type annotations from the function definitions in the provided source code."""
+
+    # Using the standard lib2to3 library to strip type annotations.
+    # Switch to another library like strip-hints is issues are found
+    from lib2to3 import fixer_base, refactor, fixer_util
+
+    class StripAnnotations(fixer_base.BaseFix):
+        PATTERN = r'''
+            typed_func_parameter=tname
+            |
+            typed_func_return_value=funcdef< any+ '->' any+ >
```

Comment:
Both libraries are not perfect with their whitespaces after the stripping (it's a non-trivial operation and the libraries have different goal here (for example, strip_hints tries to preserve all line and character positions by default (I've changed that though))).
I think we should refrain from having tests that do exact comparisons for the generated pipelines. Especially, comparing the generated python code.

The whitespaces are not supposed to be an issues for the tests since by default, lib2to3 is not used.